### PR TITLE
Set TargetFrameworkMonikerDisplayName for .NET Standard and .NET Core/.NET5+ projects.

### LINKED ISSUE #10774: .NET Core projects build with empty FrameworkDisplayName
`TargetFrameworkAttribute.FrameworkDisplayName` appears to not be set properly in .NET Core projects.
```cs
using System;
using System.Reflection;
using System.Runtime.Versioning;

class Program
{
    static void Main(string[] args)
    {
        var str = Assembly.GetExecutingAssembly().GetCustomAttribute<TargetFrameworkAttribute>().FrameworkDisplayName;
        Console.WriteLine($"\"{str}\": {str.Length}");
    }
}
```

When targeting .NET Core 3.1:
```
"": 0
```

When targeting .NET 4.8:
```
".NET Framework 4.8": 18
```

### PR DESCRIPTION
**Pull request by AraHaan**
Fixes #10774.

### CODE REVIEW

**Comment by AraHaan** [comment]
Also I think a backport of this (with changes would be needed) to .NET 5, .NET 6, and the .NET Core 3.1 SDK's if possible.

File: src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.DisableStandardFrameworkResolution.targets
Comment (dsplaisted) [inline src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.DisableStandardFrameworkResolution.targets:23]: This logic should be in terms of the `TargetFrameworkIdentifier` and `TargetFrameworkVersion` properties, and should use MSBuild built-in version comparison functions such as `VersionGreaterThanOrEquals` as appropriate so that this doesn't need to be updated for each new version of .NET.  Here is an example of similar logic for implicit compilation symbols: https://github.com/dotnet/sdk/blob/d8fac8888aca6f7365019a5d97f9658b39f9713e/src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Sdk.BeforeCommon.targets#L177-L191

For example, you could hard code the .NET Standard and .NET Core versions, and then for .NET 5 and above use something like the following:

```xml
<TargetFrameworkMonikerDisplayName Condition="''$(TargetFrameworkIdentifier)' == '.NETCoreApp' and $([MSBuild]::VersionGreaterThanOrEquals($(TargetFrameworkVersion), 5.0))">.NET $(TargetFrameworkVersion)</TargetFrameworkMonikerDisplayName>
```

**Review by dsplaisted — COMMENTED** [review]
Thanks for this PR!

It would be nice to also have some tests that verify that the description is set correctly.

File: src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.DisableStandardFrameworkResolution.targets
Comment (AraHaan) [inline src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.DisableStandardFrameworkResolution.targets:25]:
```suggestion
    <TargetFrameworkMonikerDisplayName Condition="''$(TargetFrameworkIdentifier)' == '.NETCoreApp' and $([MSBuild]::VersionGreaterThanOrEquals($(TargetFrameworkVersion), 5.0))">.NET $(TargetFrameworkVersion)</TargetFrameworkMonikerDisplayName>
```

File: src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.DisableStandardFrameworkResolution.targets
Comment (AraHaan) [reply src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.DisableStandardFrameworkResolution.targets]: Wait so there is a way to test this? Unless you are referring to a way to invoke msbuild in the unit test and checking the output ``AssemblyAttributes.cs`` file that is output from a task?

Comment (AraHaan) [reply src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.DisableStandardFrameworkResolution.targets]: After thinking about this, perhaps I should have it check .NET Core 1.0 all the way to 3.1 in the same line, then on .NET Standard 1.0 to 2.1 on a single line as well.

File: src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.DisableStandardFrameworkResolution.targets
Comment (AraHaan) [inline src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.DisableStandardFrameworkResolution.targets:21]:
```suggestion
    <TargetFrameworkMonikerDisplayName Condition="'$(TargetFrameworkIdentifier)' == '.NETStandard' and $([MSBuild]::VersionGreaterThanOrEquals($(TargetFrameworkVersion), 1.0)) and $([MSBuild]::VersionLessThanOrEquals($(TargetFrameworkVersion), 2.1))">.NET Standard $(TargetFrameworkVersion)</TargetFrameworkMonikerDisplayName>
    <TargetFrameworkMonikerDisplayName Condition="'$(TargetFrameworkIdentifier)' == '.NETCoreApp' and $([MSBuild]::VersionGreaterThanOrEquals($(TargetFrameworkVersion), 1.0)) and $([MSBuild]::VersionLessThanOrEquals($(TargetFrameworkVersion), 3.1))">.NET Core $(TargetFrameworkVersion)</TargetFrameworkMonikerDisplayName>
    <TargetFrameworkMonikerDisplayName Condition="'$(TargetFrameworkIdentifier)' == '.NETCoreApp' and $([MSBuild]::VersionGreaterThanOrEquals($(TargetFrameworkVersion), 5.0))">.NET $(TargetFrameworkVersion)</TargetFrameworkMonikerDisplayName>
```

**Comment by dsplaisted** [comment]
This looks good, but I'd like to add tests before merging it.  We may get someone on our team to do this. (FYI @marcpopMSFT)

**Comment by AraHaan** [comment]
@dsplaisted any eta on when the tests can be added to this?

**Comment by marcpopMSFT** [comment]
@gkulin you've been learning about our test infrastructure. Would you be able to help the customer out by creating a test for this?

**Comment by gkulin** [comment]
@marcpopMSFT yes, I will work on that today/tomorrow

**Comment by AraHaan** [comment]
@gkulin you still working on the tests?

**Comment by gkulin** [comment]
@AraHaan Yes, I am. I can ping you when it's finished

**Comment by gkulin** [comment]
@AraHaan I just pushed a change and added the test. We probably want to wait for an ok from either @dsplaisted or @joeloff (or both) before merging though (assuming it passes all the tests)

File: src/Tests/Microsoft.NET.Build.Tests/GivenThatWeWantToControlGeneratedAssemblyInfo.cs
Comment (joeloff) [inline src/Tests/Microsoft.NET.Build.Tests/GivenThatWeWantToControlGeneratedAssemblyInfo.cs:17]: nit: Can you sort the using statements?

File: src/Tests/Microsoft.NET.Build.Tests/GivenThatWeWantToControlGeneratedAssemblyInfo.cs
Comment (joeloff) [inline src/Tests/Microsoft.NET.Build.Tests/GivenThatWeWantToControlGeneratedAssemblyInfo.cs:768]: should `expectedVersion` be `expectedFrameworkDisplayName` based on the attribute?

**Review by joeloff — APPROVED** [review]
Overall looks good

File: src/Tests/Microsoft.NET.Build.Tests/GivenThatWeWantToControlGeneratedAssemblyInfo.cs
Comment (joeloff) [inline src/Tests/Microsoft.NET.Build.Tests/GivenThatWeWantToControlGeneratedAssemblyInfo.cs:797]: Pass in a value for the identifier parameter, you can use the `targetFrameworkVersion` parameter that will ensure the test directory hash is more unique and avoid the issues you're seeing in Helix I think

**Comment by AraHaan** [comment]
Thanks, also is this good to merge now?

File: src/Tests/Microsoft.NET.Build.Tests/GivenThatWeWantToControlGeneratedAssemblyInfo.cs
Comment (gkulin) [reply src/Tests/Microsoft.NET.Build.Tests/GivenThatWeWantToControlGeneratedAssemblyInfo.cs:772]: @AraHaan it warns that .NET core 2.1 is out of support.

Comment (gkulin) [reply src/Tests/Microsoft.NET.Build.Tests/GivenThatWeWantToControlGeneratedAssemblyInfo.cs:772]: we're just suppressing it here because we want to strictly check the output from Program.cs

**Review by dsplaisted — APPROVED** [review]

**Comment by AraHaan** [comment]
Should a milestone be added to this and the issue this fixed (the milestone changed from future)?

**Pull request merged**
